### ISSUE TITLE
Slow session set up

### ISSUE THREAD

**Issue by hazeyb** [rtbrick/bngblaster#74]
**Describe the bug**

When setting up around 100 sessions, the session set up is very slow;

Sessions             100 (100 PPPoE / 0 IPoE)
  Established         51 [###############################                             ]
  Outstanding        800 [############################################################]
  Terminated           0 [                                                            ]
  Setup Time       13400 ms
  Setup Rate        3.81 CPS (MIN: 3.42 AVG: 4.02 MAX: 4.57)
  Flapped            823
 
If I run a PCAP, and filter for a particular vlan (one that hasn't yet auth'd successfully).   I can see that PPPOE has established, and PPP LCP has begun.  Following the PPP CHAP challenge from our BNG bngblaster immediately responds with a PPP LCP terminate request.  We then get the Termination ACK, PPPOE terminates, and the entire process starts again.  

We see that repeat over and over.  Though eventually the other sessions get established.
Sessions             100 (100 PPPoE / 0 IPoE)
  Established         61 [#####################################                       ]
  Outstanding        800 [############################################################]
  Terminated           0 [                                                            ]
  Setup Time      198400 ms
  Setup Rate        0.31 CPS (MIN: 0.31 AVG: 2.57 MAX: 4.57)
  Flapped           1000

We have tested the BNG with the same VLANs from a Spirent and set up of around 2000 sessions takes 10-20 seconds.

**To Reproduce**

Version (`bngblaster -v`):

```
Version: 0.7.1
IO Modes: packet_mmap_raw (default), packet_mmap, raw
```

JSON configuration:
```json
{
    "interfaces": {
        "access": [
            {
                "interface": "enp3s0f0",
                "type": "pppoe",
		"outer-vlan": 4,
                "inner-vlan-min": 1,
                "inner-vlan-max": 4094,
                "authentication-protocol": "CHAP"
            }
        ]
    },
    "sessions": {
        "count": 100,
	"session-time": 0,
        "max-outstanding": 800,
        "start-rate": 400,
        "stop-rate": 400
    },
    "pppoe": {
	"service-name": "access",
        "reconnect": true,
        "discovery-timeout": 100,
        "discovery-retry": 1,
	"host-uniq": true,
        "vlan-priority": 6
    },
    "ppp": {
        "mru": 1492,
        "authentication": {
            "username": "mx10k3test{session-global}@zen",
            "password": "password",
            "timeout": 5,
            "retry": 30,
	    "protocol": "CHAP"
        },
        "lcp": {
            "conf-request-timeout": 5,
            "conf-request-retry": 10,
            "keepalive-interval": 30,
            "keepalive-retry": 3
        },
        "ipcp": {
            "enable": true,
            "request-ip": true,
            "request-dns1": true,
            "request-dns2": true,
            "conf-request-timeout": 1,
            "conf-request-retry": 10
        }
    },
    "access-line": {
        "agent-remote-id": "DEU.RTBRICK.{session-global}",
        "agent-circuit-id": "0.0.0.0/0.0.0.0 eth 0:{session-global}",
        "rate-up": 1024,
        "rate-down": 16384

```

1. sudo bngblaster -C pppoe.json -I

**Expected behavior**

We'd expect the session setup to be quite quick. PPPOE set up completes in a few seconds, but we don't understand why bngblaster is sending the PPP LCP terminate requests to the BNG.  If we wait long enough, all the sessions will eventually come up. 

**Screenshots**

![image](https://user-images.githubusercontent.com/11196107/163248006-cf73407d-e4df-4aa9-8eec-33fde40ccc10.png)

  Access Interface Protocol Packet Stats:
    ARP    TX:          0 RX:          0
    PADI   TX:        902 RX:          0
    PADO   TX:          0 RX:        902
    PADR   TX:        902 RX:          0
    PADS   TX:          0 RX:        902
    PADT   TX:        839 RX:         63
    LCP    TX:       2986 RX:       2970
    PAP    TX:          0 RX:          0
    CHAP   TX:         64 RX:        966
    IPCP   TX:        192 RX:        192
    IP6CP  TX:        129 RX:        129
    IGMP   TX:          0 RX:          0
    ICMP   TX:          0 RX:          0
    DHCP   TX:          0 RX:          0
    DHCPv6 TX:          0 RX:          0
    ICMPv6 TX:         64 RX:        169
    IPv4 Fragmented       RX:          0

  Access Interface Protocol Timeout Stats:
    LCP Echo Request:          0
    LCP Request:               0
    IPCP Request:              0
    IP6CP Request:             0
    PAP:                       0
    CHAP:                      0
    DHCP Request:              0
    DHCPv6 Request:            0

**Additional context**

**Comment by GIC-de** [comment]
Today, the BNG Blaster does not support CHAP challenges with other lengths than 16 bytes but this is something I could fix.

**Comment by GIC-de** [comment]
I added a log message (6d240c7ab5f483e6f14240be386f449ec8987548) for this case and will address to support variable CHAP lengths for the next release.

**Comment by GIC-de** [comment]
I just merged a change into `dev` branch which should allow all CHAP lengths.

**Comment by hazeyb** [comment]
I've installed from the dev branch;

zenadmin@bngblaster01:~$ sudo bngblaster -v
Version: DEV
Compiler: GNU (9.4.0)
GIT:
  REF: dev
  SHA: 4de1019d533b4b0faef22c4e39a9c9350a19baca
IO Modes: packet_mmap_raw (default), packet_mmap, raw

I've left the BNG (Juniper MX10k3) to use a range of challenge lengths, and its all connecting fine now.  CPS rate is as expected;

Sessions           12000 (12000 PPPoE / 0 IPoE)
  Established       2848 [###############                                             ]
  Outstanding       1965 [########################################################### ]
  Terminated           0 [                                                            ]
  Setup Time       30002 ms
  Setup Rate       93.76 CPS (MIN: 23.00 AVG: 95.09 MAX: 107.18)
  Flapped              5

**Comment by GIC-de** [comment]
Thanks for you feedback. Fix is also included in latest version 0.7.11!
https://github.com/rtbrick/bngblaster/releases/tag/0.7.11
